libraries/AP_Logger/LogFile.cpp

```diff
@@ -500,6 +500,21 @@ void AP_Logger::Write_POS(AP_AHRS &ahrs)
     WriteBlock(&pkt, sizeof(pkt));
 }
 
+// Write a VEL packet
+void AP_Logger::Write_VEL(AP_AHRS &ahrs)
+{
+    Vector3f vel;
+    struct log_VEL pkt = {
+        LOG_PACKET_HEADER_INIT(LOG_VEL_MSG),
+        time_us : AP_HAL::micros64(),
+        vx      : ahrs.get_velocity_NED(vel) ? vel.x : quiet_nanf(),
```

Comment:
Calling this method three times seems a little much :-)
Probably want a `bool have_vel = ahrs.get_velocity_NED(vel);` outside the struct definition.

Reply:
Yeah, so nice. I thought it was not good but didn't get better idea to do it.